src/containers/WarningConfirmThresholdScenario/index.jsx

```diff
@@ -49,28 +46,48 @@ class WarningConfirmThresholdScenario extends React.Component {
 		if (!isValid) {
 			return;
 		}
+		const permissionsKeys = this.props.form.toJS();
+		const activeAccounts = this.props.activeAccounts.toJS();
 		const nextTreshold = this.props.treshold.value;
-		let { permissionsKeys } = this.props;
-		const { network, account } = this.props;
-		const accountId = account.toJS().id;
-		permissionsKeys = permissionsKeys.toJS();
-		const activeKeysData = {
-			keys: permissionsKeys.active.keys.concat(permissionsKeys.active.accounts),
-			threshold: permissionsKeys.active.threshold,
-		};
-		const maxNextThreshold = activeKeysData.keys
-			.reduce((accumulator, currentKey) => accumulator + currentKey.weight, 0);
+		const { keys, accounts } = permissionsKeys.active;
+		let goodNextThreshold = 0;
+		let maxNextThreshold = 0;
+		for (const key in keys) {
+			maxNextThreshold += (+keys[key].weight.value);
+			if (keys[key].hasWif && keys[key].hasWif.value) {
+				goodNextThreshold += (+keys[key].weight.value);
+			}
+		}
+		for (const account in accounts) {
+			maxNextThreshold += (+accounts[account].weight.value);
+			for (const activeAccount in activeAccounts) {
+				if (activeAccount === account) {
+					goodNextThreshold += (+accounts[account].weight.value);
+				}
+			}
+		}
+		const { echoRand } = permissionsKeys;
+		for (const key in echoRand.keys) {
+			if (!echoRand.keys[key].hasWif.value) {
+				this.setState({
```

Comment:
make 2 warning message for active part and echoRand part instead of overwrite them